src/vm/errors/vm_errors.rs

```diff
@@ -143,4 +143,6 @@ pub enum VirtualMachineError {
     NegBuiltinBase,
     #[error("Security Error: Invalid Memory Value: temporary address not relocated: {0}")]
     InvalidMemoryValueTemporaryAddress(Relocatable),
+    #[error(transparent)]
+    Other(#[from] anyhow::Error),
```

Comment:
Can we use `Box<dyn Error>` here?

Reply:
Yes we can. Why do you prefer it?

Reply:
https://docs.rs/anyhow/latest/anyhow/struct.Error.html
Anyhow provide a few more interesting features in comparison, especially requiring the error type to be Send and Sync.
What is the rationale for not wanting those?

Reply:
I dont think it would be a good idea to force users to use the `anyhow` crate if they want to use their custom error types. If `Send` and `Sync` are a necessity, we can always use `Box<dyn Error + Send + Sync>`.

Reply:
Alright. @hurrikaanig is finishing some work for Kakarot and will come back to this one really soon